### PR TITLE
added maxAwaitTimeMS to the CRUD specification.

### PR DESCRIPTION
**Pull request by craiggwilson**
cc @jyemin, @behackett, @christkv

[Review on Reviewable](https://reviewable.io/reviews/mongodb/specifications/71)

### CODE REVIEW

**Comment by christkv** [comment]
lgtm

**Comment by behackett** [comment]
Can we provide any guidance for how this should be used? I worry that this is going to be another confusing knob with no guidance.

**Comment by craiggwilson** [comment]
@behackett I would say yeah, but that it's not the place of CRUD spec to be doing that. This falls exactly in the spot that the find/getmore spec is attempting to fill.

Just to keep this all here, why type of guidance are you thinking? I'm my eyes, this is pretty simple. When the cursor is a tailable_await cursor, this is used to specify maxTimeMS on getMores. That's it.

**Comment by behackett** [comment]
What I mean is, what value should a user use? The same value as maxTimeMS? A lower value? A greater value? Are there reasons to not user the feature, or is it a MUST when using await cursors? We have to provide some guidance or the option is worse than having no option at all.

**Comment by craiggwilson** [comment]
Ah... Good question. @christkv, I think this is more your area of expertise. The default (when left unspecified) will be 1 second I believe. I would think that the right amount of time is going to be a balance between how often inserts are happening in the server and how much time you want to be waiting on the server vs. the client.  More so than normal, I think the answer is going to be it depends. However, the number of people who use tailable cursors is a vast minority.

Ulimately, adding this new knob isn't confusing anything that wasn't already confused. If we need guidance now, we should have had guidance before as well.

**Comment by behackett** [comment]
> If we need guidance now, we should have had guidance before as well.

I agree, but adding a specific option makes things worse by signaling that you really need to use it, yet providing no guidance about using it. :-)

**Comment by craiggwilson** [comment]
Where do you propose we put it? I'd say it belongs in the server's docs under the getMore documentation, but that would be completely hidden.

**Comment by behackett** [comment]
It should likely live in each driver's docs for the option. But since we likely want each driver to document it the same, we have to have someplace to copy it from. Here seems no better or worse than anywhere else.

**Comment by craiggwilson** [comment]
Well, I guess that's fine.  We need some actual guidance on it. @christkv has been around this a lot longer...

**Comment by xdg** [comment]
Could we please move this conversation to a SPEC ticket so it stays on the record (and doesn't notify everyone following this repo)?

File: source/crud/crud.rst [hunk 1]
```diff
@@ -362,8 +362,21 @@ Read
     limit: Optional<Int32>;
 
     /**
+     * The maximum amount of time for the server to wait on new documents to satisfy a tailable cursor 
+     * query. This only applies to a TAILABLE_AWAIT cursor. When the cursor is not a TAILABLE_AWAIT cursor,
+     * this option is ignored.
+     *
+     * On servers >= 3.2, this option will be specified on the getMore command as "maxTimeMS". The default
+     * is no value: no "maxTimeMS" is sent to the server with the getMore command.
```
Comment (jmikola) [inline source/crud/crud.rst]: Can we change "no value: no maxTimeMS is..." to either "no value (i.e. no maxTimeMS is..." or "no value, which means that no maxTimeMS is"?

**Comment by jmikola** [comment]
Closing this, as the change was merged in https://github.com/mongodb/specifications/commit/bb22f121ab0e105e1ccd1786545559b546d490b1.

For reference, conversation about the option was continued in [DRIVERS-272](https://jira.mongodb.org/browse/DRIVERS-272).

**Pull request closed without merging**